### PR TITLE
allow editing single files shared as public link

### PR DESCRIPTION
**Pull request by schiessle**
allow editing single files shared as public link

cc @blizzz

### CODE REVIEW

**Review by blizzz — APPROVED** [review]
works

File: apps/files_sharing/lib/Controller/ShareAPIController.php [hunk 1]
```diff
@@ -668,6 +668,7 @@ public function updateShare(
 					\OCP\Constants::PERMISSION_READ | \OCP\Constants::PERMISSION_CREATE | \OCP\Constants::PERMISSION_UPDATE, // legacy
 					\OCP\Constants::PERMISSION_READ | \OCP\Constants::PERMISSION_CREATE | \OCP\Constants::PERMISSION_UPDATE | \OCP\Constants::PERMISSION_DELETE, // correct
 					\OCP\Constants::PERMISSION_CREATE, // hidden file list
+					\OCP\Constants::PERMISSION_READ | \OCP\Constants::PERMISSION_UPDATE, // allow to edit single files
```
Comment (nickvergessen) [inline apps/files_sharing/lib/Controller/ShareAPIController.php]: I remember something with "update is delete+create on dav".
So this needs to be tested with the problematic app before merging....

**Review by nickvergessen — CHANGES_REQUESTED** [review]

File: apps/files_sharing/lib/Controller/ShareAPIController.php (hunk 1 above)
Comment (schiessle) [inline apps/files_sharing/lib/Controller/ShareAPIController.php]: Seems that the this is not the case for Collabora, @LukasReschke can you test it? Thanks!

**Comment by nickvergessen** [comment]
Just to be sure I shall try to smash this

**Comment by nickvergessen** [comment]
Smashbox passed ✅

**Review by icewind1991 — APPROVED** [review]

File: apps/files_sharing/lib/Controller/ShareAPIController.php (hunk 1 above)
Comment (LukasReschke) [inline apps/files_sharing/lib/Controller/ShareAPIController.php]: Collabora doesn't go via DAV but the nodes API directly.

**Pull request merged**